sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/ShareAsyncApiTests.java

```diff
@@ -261,9 +274,9 @@ public void createIfNotExistsShareWithInvalidArgs(Map<String, String> metadata,
 
     @Test
     public void deleteShare() {
-        primaryShareAsyncClient.create().block();
-        StepVerifier.create(primaryShareAsyncClient.deleteWithResponse())
-            .assertNext(it -> FileShareTestHelper.assertResponseStatusCode(it, 201));
+        StepVerifier.create(primaryShareAsyncClient.create().then(primaryShareAsyncClient.deleteWithResponse()))
+            .assertNext(it -> FileShareTestHelper.assertResponseStatusCode(it, 201))
```

Comment:
It looks like as the original test never had `.verifyComplete()`, it was never caught that the correct status code is 202 and not 201. You can change this line to ` .assertNext(it -> FileShareTestHelper.assertResponseStatusCode(it, 202))`